lib/zoho_sign/response_handler.rb

```diff
@@ -3,47 +3,56 @@
 module ZohoSign
   # Internal class to centralize the code related to the response of the Zoho Sign API
   class ResponseHandler
-    def initialize(params)
-      @params = params || {}
+    def initialize(response, http_response = nil)
+      @response = response || {}
+      @http_response = http_response
     end
 
     def body
-      @params
+      @response
     end
 
     def data(key)
-      @params[key]
+      return unless @response.is_a?(Hash)
+
+      @response[key]
     end
 
     def success?
-      @params[:status] == "success"
+      return false unless @response.is_a?(Hash)
+
+      @response[:status] == "success"
     end
 
     def error?
-      @params[:status] == "error" || @params[:status] == "failure"
+      return false if @http_response == 200
+
+      (!@http_response.nil? && @http_response != 200) ||
+        @response[:status] == "error" ||
+        @response[:status] == "failure"
```

Comment:
@http_response >= 400

Reply:
same than the comment before, the http_response is an optional params so sometimes we don't have it and then we seek for @response[:status]